app/baseComponents/SideMenu/SideMenu.js

```diff
@@ -0,0 +1,254 @@
+`;
+
+// $FlowStyledIssue
+const StyledTopOrBottomWrapper = styled.div`
+  border-top: 1px solid ${({ topOrBottom }) => (topOrBottom === 'bottom' ? smColors.borderGray : 'transparent')};
+`;
+
+// $FlowStyledIssue
+const StyledMenuEntryWrapper = styled(StyledAction)`
+  cursor: ${({ disabled }) => (disabled ? 'default' : 'pointer')};
+  opacity: ${({ disabled }) => (disabled ? 0.5 : 1)};
+  pointer-events: ${({ disabled }) => (disabled ? 'none' : 'all')};
+`;
+
+// $FlowStyledIssue
+const StyledMenuEntry = styled.div`
+  width: ${ENTRY_WIDTH}px;
+  height: ${ENTRY_HEIGHT}px;
+  cursor: pointer;
+  display: flex;
+  flex-direction: row;
+  justify-content: flex-start;
+  div :first-child {
+    width: 12px;
+    background-color: ${({ isSelected }) => (isSelected ? smColors.green : smColors.white)};
+  }
+`;
+
+const StyledIconWrapper = styled.div`
+  height: 24px;
+  width: 24px;
+  position: relative;
+`;
+
+const StyledIcon = styled.img`
+  height: 80%;
+  width: 80%;
+  position: absolute;
+  top: 17px;
+  left: 6px;
+`;
+
+const StyledLabel = styled.span`
+  font-family: ${smFonts.fontLight12.fontFamily};
+  font-size: ${smFonts.fontLight12.fontSize}px;
+  font-weight: ${smFonts.fontLight12.fontWeight};
+`;
+
+const StyledLabelWrapper = styled.div`
+  padding-top: 16px;
+  padding-left: 24px;
+  user-select: none;
+`;
+
+const StyledLoadingIconWrapper = styled.div`
+  position: relative;
+`;
+
+export default class SideMenu extends React.Component<SideMenuProps, SideMenuState> {
+  constructor(props: SideMenuProps) {
+    super(props);
+    this.state = {
+      isOpen: !!props.openOnInit,
+      selectedId: -1,
+      width: props.openOnInit ? ENTRY_WIDTH : ENTRY_WIDTH_CLOSED
+    };
+  }
+
+  render() {
+    const { width, isOpen } = this.state;
+
+    return (
+      <StyledSideMenuRoot width={width}>
+        <StyledToggleWrapper onClick={this.handleSideMenuToggle}>
+          <StyledToggle isOpen={isOpen} width={width} src={openIcon} alt="Icon missing" />
+        </StyledToggleWrapper>
+        {this.renderMenuElem()}
+      </StyledSideMenuRoot>
+    );
+  }
+
+  renderMenuEntryInnerElem = (entry: SideMenuEntry) => {
+    const { isOpen, selectedId } = this.state;
+    const { loadingEntry } = this.props;
+    return (
+      <StyledMenuEntry isSelected={entry.id === selectedId}>
+        <div />
+        <StyledIconWrapper>
+          <StyledIcon src={entry.iconSrc} alt="Icon missing" />
+        </StyledIconWrapper>
+        <StyledLabelWrapper>
+          <StyledLabel>{entry.label}</StyledLabel>
+        </StyledLabelWrapper>
+        <StyledLoadingIconWrapper>
+          {loadingEntry && loadingEntry.id === entry.id && (
+            <SmallLoader isLoading={loadingEntry.isLoading} loadingLeft={isOpen ? 8 : -96} loadingSize={isOpen ? 18 : 25} loadingTop={isOpen ? 18 : 14} />
+          )}
+        </StyledLoadingIconWrapper>
+      </StyledMenuEntry>
+    );
+  };
+
+  renderTopOrBottomElem = (topOrBottom: 'top' | 'bottom') => {
```

Comment:
make this func same style as next one (immediate return)

Reply:
done